pulsar-tracing/src/main/java/com/datastax/oss/pulsar/tracing/BrokerTracing.java

```diff
@@ -0,0 +1,645 @@
+  //
+  //        // todo: how to read topic props
+  //        return true;
+  //    }
+
+  /* ***************************
+   **  Administrative events
+   ******************************/
+
+  public void onConnectionCreated(ServerCnx cnx) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    log.info("Connection created: {}", traceDetails);
+  }
+
+  public void producerCreated(ServerCnx cnx, Producer producer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateProducerDetails(level, producer, traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Producer created: {}", traceDetails);
+  }
+
+  public void producerClosed(ServerCnx cnx, Producer producer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateProducerDetails(level, producer, traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Producer closed: {}", traceDetails);
+  }
+
+  public void consumerCreated(ServerCnx cnx, Consumer consumer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateConsumerDetails(level, consumer, traceDetails);
+    populateSubscriptionDetails(level, consumer.getSubscription(), traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Consumer created: {}", traceDetails);
+  }
+
+  public void consumerClosed(ServerCnx cnx, Consumer consumer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateConsumerDetails(level, consumer, traceDetails);
+    populateSubscriptionDetails(level, consumer.getSubscription(), traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Consumer closed: {}", traceDetails);
```

Comment:
we should discuss with them about the format, maybe we have to JSON encode the details ? I am wondering about having he ability to parse the contents, unfortunately Map.toString() doesn't enforce a format and it doesn't perform any kind of escaping.